src/main/java/org/example/repository/QuestionRepositoryImpl.java

```diff
@@ -0,0 +1,117 @@
+package org.example.repository;
+
+import org.example.model.Question;
+import org.example.repository.dao.QuestionRepository;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QuestionRepositoryImpl implements QuestionRepository {
+
+    private Connection connection;
+
+    public QuestionRepositoryImpl(Connection connection) {
+        this.connection = connection;
+    }
+
+    private String findByID = """ 
+            select * from questions where id = ?
+            """;
+
+    private String findByTopic = """
+            select * from questions where topic = ?
+            """;
+
+    private String add = """
+            insert into questions(text, topic)
+            values(? , ?);
+            """;
+
+    private String removeById = """
+            delete from questions where id = ? 
+            """;
+
+    private String updateTopic = """
+            update questions set topic = ? where id = ?
+            """;
+
+    @Override
+    public Question get(int id) {
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(findByID);
+            preparedStatement.setInt(1, id);
+            preparedStatement.execute();
+            ResultSet resultSet = preparedStatement.getResultSet();
+            resultSet.next();
+            return Question.builder()
+                    .id(resultSet.getInt("id"))
+                    .text(resultSet.getString("text"))
+                    .topic(resultSet.getString("topic"))
+                    .build();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```

Comment:
it is better to at least describe in the console what went wrong and then throw the exception.  And it will be even better to create and throw your own exception. This comment is related to all RuntimeException that you throw